src/components/StrategicWords.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import data from './data.json';
+
+export default function StrategicWords({ jsonData, arrayStrategic }) {
+
+    let jsonDataWords = jsonData.map(item => item.word);
+    let jsonDataLetters = [];
+    for (let i = 0; i < jsonDataWords.length; i++) {
+        for (let j = 0; j < jsonDataWords[i].length; j++) {
+            let a = jsonDataWords[i];
+            jsonDataLetters.push(a[j]);
+        }
+    }
+
+    let jsonDataLettersCounts = jsonDataLetters.reduce(function (acc, curr) {
+        return acc[curr] ? ++acc[curr] : acc[curr] = 1, acc;
+    }, {});
+
+    let jsonDataLettersCountsSorted = Object.fromEntries(
+        Object.entries(jsonDataLettersCounts).sort(([, a], [, b]) => a - b)
+    );
+
+    let filteredMasterData = [...data];
+    let x = Object.keys(jsonDataLettersCountsSorted)[0];
+
+    for (let i = 0; i < 2; i++) {
+        filteredMasterData = filteredMasterData.filter(item => item.word.includes(Object.keys(jsonDataLettersCountsSorted)[i]));
+    }
+
+    // when it drops below 10
+    const final = () => {
+        if (jsonData.length === 1) {
+            return jsonData;
+        } else if (jsonData.length > 1 && jsonData.length < 10) {
+            return filteredMasterData;
+        } else if (arrayStrategic.length === 0) {
+            return filteredMasterData;
+        } else {
+            return arrayStrategic;
+        }
+    };
+
+    const words = () => {
+        let x = final().map(item => {
+            return <li className='strategic-words'>{item.word}</li>;
+        });
+
+        let y = x.slice(-5);
```

Comment:
a lot of random numbers in this file. Are these word lengths? Word lengths * 2? Are they related? Derived from each other? 

I recommend defining some well named constants at the top of this file, splitting out the functionality and commenting this. I have no idea if it works.

Also on that note, I'd probably go ahead and pull it into a util file and use jest for some rudimentary test cases.

https://jestjs.io/

Reply:
This is a good opportunity for you to test the limits of your functionality also. Can you think of ways this might break? Add it as a test case.